2996_reflection/reflection.md

```diff
@@ -3386,14 +3393,17 @@ Add a new paragraph between [expr.eq]{.sref}/5 and /6:
-* [*.#]{.pnum} Otherwise, if both operands are reflections of a base class specifier, then they compare equal if they are reflections of the same base class specifier.
-* [*.#]{.pnum} Otherwise, if one operand is a reflection of a base class specifier and the other is not, then they compare unequal.
-* [*.#]{.pnum} Otherwise, if neither operand is a reflection of a value, then they compare equal if they are reflections of the same entity.
-* [*.#]{.pnum} Otherwise, if one operand is a reflection of a value and the other is not, then they compare unequal.
-* [*.#]{.pnum} Otherwise, if both operands are reflections of values, then they compare equal if and only if the reflected values are _template-argument-equivalent_ ([temp.type]{.sref}).
-* [*.#]{.pnum} Otherwise the result is unspecified. [For example, if the operands are results of calls to `data_member_spec`]{.note}
+* [*.#]{.pnum} Otherwise, if both operands represent entities, then the compare equal if and only if they are reflections of the same entity.
+* [*.#]{.pnum} Otherwise, if one operand represents an entity, then they compare unequal.
+* [*.#]{.pnum} Otherwise, if both operands represent variables, then the compare equal if and only if they are reflections of the same entity.
+* [*.#]{.pnum} Otherwise, if one operand represents a variable, then they compare unequal.
+* [*.#]{.pnum} Otherwise, if one operand represents a namespace alias or type alias and the other operand does not represent the same kind of alias, they compare unequal. [A reflection of a type and a reflection of an alias to that same type do not compare equal.]{.note}
+* [*.#]{.pnum} Otherwise, if both operands represent namespace aliases or type aliases, then they compare equal if and only if their represented aliases share the same name, are declared within the same enclosing scope, and alias the same underlying entity.
+* [*.#]{.pnum} Otherwise, if both operands represent base class specifiers, then they compare equal if and only if they represent the same base class specifier.
+* [*.#]{.pnum} Otherwise, if one operand represents a base class specifier, then they compare unequal.
+* [*.#]{.pnum} Otherwise, if both operands represent values, then they compare equal if and only if the reflected values are _template-argument-equivalent_ ([temp.type]{.sref}) and share the same type.
+* [*.#]{.pnum} Otherwise, if one operand represents a value, then they compare unequal.
+* [*.#]{.pnum} Otherwise, both operands `O@~_1_~@` and `O@~_2_~@` represent descriptions of declarations of non-static data members: Let `C@~_1_~@` and `C@~_2_~@` be invented class types such that each `C@~_k_~@` has a single non-static data member having the properties described by `O@~_k_~@`. The operands compare equal if and only if the data members of `C@~_1_~@` and `C@~_2_~@` would  share the same type, name (if any), `$alignment-specifiers$` (if any), width, and attributes.
```

Comment:
What does introducing the class buy us? Can't we just directly say that they compare equal if they share the same type, name, alignment, width, and attributes?

Reply:
I'm trying to dance around a few concerns here:
1. There's no way to introspect certain properties of a `data_member_spec` (for example: whether it has `no_unique_address` attribute), so it feels a bit odd to directly discuss "which properties" they have.
2. Neither do I want to specify "if it was created with a call to `data_member_spec` having the same arguments...": e.g., arguably `data_member_spec(^int, {.name="m"})` should compare equal to `data_member_spec(^int, {.name=u8"m"})`.

Reply:
Ok that makes sense. We'll see what CWG says I guess.